### PR TITLE
#161967011 Implements password reset

### PR DESCRIPTION
**Pull request by gwako94**
#### What does this PR do?
Implements password reset using email

#### Description of Task to be completed?
Implements password reset using email
write tests for email reset

#### How should this be manually tested?

```
1. $ git clone https://github.com/andela/ah-jumanji.git
2. $ git checkout ft-reset-password-161967011
3. $ cd into ah-jumanji
4. $ virtualenv -p python3 venv
5. $ source venv/bin/activate
6. (venv) $ pip install -r requirements.txt
7. (venv) $ echo "DATABASE_URL=postgres://username:password@localhost/db_name" >> .env
8. (venv) $ python manage.py migrate
9. (venv) $ python manage.py runserver
```
Fire up [Postman](https://www.getpostman.com/) and test the following endpoints:
```
1: http://127.0.0.1:8000/api/users/reset_password

```
With the following payload:
```
{
    "email": "your-email"
}
```

Reset url will come with a token
```
2: http://127.0.0.1:8000/api/users/reset_password/<token>

```
With the following payload:
```
{
    "password": "new_password"
}
```

#### What are the relevant pivotal tracker stories?
[#161967011](https://www.pivotaltracker.com/story/show/161967011)

#### Screenshots (if appropriate)
screenshot 2018-12-05 at 19 52 53

screenshot 2018-12-06 at 11 31 33

#### Checklist:

- [x] My code follows the style guidelines of this project
- [x] At least 2 people have reviewed my PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] My PR has one commit.

### CODE REVIEW

**Comment by mashafrancis** [comment]
The functionality works awesome. I have tested it. Good work. 
Pending issues with our Hound barking to be fixed, then you'll be good.

**Comment by gwako94** [comment]
> The functionality works awesome. I have tested it. Good work.
> Pending issues with our Hound barking to be fixed, then you'll be good.

On it

**Review by mashafrancis — APPROVED** [review]
Good work.

**Review by ghost — APPROVED** [review]
This is ready to merge cc @mirr254

**Review by EmmanuelBeja — APPROVED** [review]
Nice work.

**Review by EmmanuelBeja — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -24,6 +24,9 @@ var/
 .installed.cfg
 *.egg
 .idea/
+.DS_Store
```
Comment (gitau-kinuthia) [inline .gitignore:27]: .DS_Store and .vscode entries are redundant in .gitignore

**Review by gitau-kinuthia — APPROVED** [review]
The functionality is well implemented. However, your .gitignore has redundant entries for .vscode and .DS_store

**Pull request merged**